### ISSUE TITLE
Bad conditions coverage with when string used

### ISSUE THREAD

**Issue by ChristopheCiambella** [Kotlin/kotlinx-kover#276]
**Describe the bug**
Kover does not correctly interpret conditions coverage with when strings.

**Expected behavior**
When all case of when is being tested, the final coverage must be 100%

**Reproducer**
I've been wrote a Stackoverflow question with all detailed things about this issue: https://stackoverflow.com/questions/74387415/why-is-android-kover-not-covering-my-string-when-statements

**Environment**
 - Kover Gradle Plugin version: 0.6.1
 - Gradle version: 7.2.1
 - Kotlin project type: Kotlin/Android

**Comment by zuevmaxim** [comment]
@ChristopheCiambella Thank you for your report!
Here, Kotlin compiler generates a switch on `String.hashCode` and also checks for equality with an `if`. These `if` branches are uncovered - they should be ignored by the coverage engine

**Comment by zuevmaxim** [comment]
Fixed in agent version `1.0.690`

To use custom engine you need to configure it in all projects where Kover is used

for Kotlin Gradle script
```
kover {
    engine.set(kotlinx.kover.api.IntellijEngine("1.0.690"))
}
```
for Groovy
```
kover {
    engine = new kotlinx.kover.api.IntellijEngine("1.0.690")
}
```

**Comment by ChristopheCiambella** [comment]
@zuevmaxim I confirm that your fix work very well on my project :)

Thanks for work !

I'm waiting to the final release to definitively implement it.

**Comment by shanshin** [comment]
Fixed in `0.7.0-Alpha`
